python/mxnet/gluon/contrib/data/audio/transforms.py

```diff
@@ -0,0 +1,208 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+# coding: utf-8
+# pylint: disable= arguments-differ
+"Audio transforms."
+
+import warnings
+import numpy as np
+try:
+    import librosa
+except ImportError as e:
+    warnings.warn("gluon/contrib/data/audio/transforms.py : librosa dependency could not be resolved or \
+    imported, could not provide some/all transform.")
+
+from ..... import ndarray as nd
+from ....block import Block
+
+class MFCC(Block):
+    """
+        Extracts Mel frequency cepstrum coefficients from the audio data file
+        More details : https://librosa.github.io/librosa/generated/librosa.feature.mfcc.html
+
+    Parameters
+    ----------
+    Keyword arguments that can be passed, which are utilized by librosa module are:
+    sr: int, default 22050
+        sampling rate of the input audio signal
+    n_mfcc: int, default 20
+        number of mfccs to return
+
+
+    Inputs:
+        - **x**: input tensor (samples, ) shape.
+
+    Outputs:
+        - **out**: output array is a scaled NDArray with (samples, ) shape.
+
+    """
+
+    def __init__(self, **kwargs):
+        self.kwargs = kwargs
+        super(MFCC, self).__init__()
+
+    def forward(self, x):
+        if not librosa:
```

Comment:
this check is already done at the top

Reply:
Yes. Not all the transforms require librosa like Scaling and Pad/Trim. so, testing the availability of librosa in the methods that actually use it and warning the user that this particular transform cannot be applied. What do you recommend?